src/analyses/memOutOfBounds.ml

```diff
@@ -502,6 +433,66 @@ struct
     Option.iter (fun x -> check_lval_for_oob_access ctx x) lval;
     ctx.local
 
+  let query ctx (type a) (q: a Queries.t): a Queries.result =
+    match q with
+    | Queries.AllocMayBeOutOfBounds (e, i, o, t) ->
+      begin match i with 
+        | i when not @@ ID.is_bot i  -> 
```

Comment:
Otherwise, the result is top? Then this may be written as a `when` clause in the match?

Reply:
Yes you are correct. I have rewritten this in a3af62b0d06c5a4ef857b827d811392b12845a70.